Overhaul list display of mods + add download counts and external links

lporu opened the pull request:
Some small changes to the way mods are displayed in a list to make it easier to edit in manual mode, by adding external links to the original mod pages and adding a download count to all mods. Inspired by the way modrinth displays the mods on their site.

![Screenshot 2022-10-11 at 13 43 31](https://user-images.githubusercontent.com/116395651/197340651-fa19ba1e-1a08-47f1-a1a9-8251b629fad1.png)

Signed-off-by: lporu <116395651+lporu@users.noreply.github.com>

ryanccn commented on src/components/partials/RichModDisplay.tsx:
```diff
@@ -41,50 +17,72 @@ interface Props {
+        <div className="shrink-0">
+          {data.iconUrl && (
+            <img
+              src={data.iconUrl}
```
might want to add `width` and `height` attributes here to prevent layout shift

ryanccn commented on src/components/partials/RichModDisplay.tsx:
```diff
@@ -41,50 +17,72 @@ interface Props {
+              </a>
+            </div>
+          </div>
+          <div className="min-w-fit text-right">
```
on mobile this is still rendered right to the description, resulting in it being a bit squished; could you move it to be displayed under the description (like Modrinth does) on mobile?

ryanccn reviewed (changes requested):
Overall great stuff, just a few little changes and it's good to go! Thanks 😃

lporu commented:
Moved the download count + button on smaller screens (mobile) and added the `width` and `height` attributes to mod icons. Should now feel much better on mobile screens.

ryanccn reviewed (approved):
Looks great!
